tests/data/test_grids.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+
+import numpy as np
+from toshi_hazard_store.model import ProbabilityEnum
+
+from nzshm_hazlab.data import HazardGrids
+
+
+class DummyGridLoader:
+    """A dummy grid loader class."""
+
+    def get_grid(self, hazard_model_id, imt, grid_name, vs30, poe, agg: str):
+        """A method for the dummy loader."""
+        filename = f"hazard_grid-{hazard_model_id}-{imt}-{grid_name}-{vs30}-{poe.name}-{agg}.npy"
+        filepath = Path(__file__).parent.parent / f"fixtures/data/grids/{filename}"
+        return np.load(filepath)
+
+
+def test_get_grid():
+    hazard_model_id = "NSHM_v1.0.4"
+    grid_name = "NZ_0_1_NB_1_1"
+    poe = ProbabilityEnum._2_PCT_IN_50YRS
+    agg = "mean"
+    imt = "SA(1.0)"
+    vs30 = 750
+
+    loader = DummyGridLoader()
+    hazard_grids = HazardGrids(loader=loader)
+
+    imtls = hazard_grids.get_grid(hazard_model_id, imt, grid_name, vs30, poe, agg)
+    assert imtls.shape == (3741,)
+
+
+def test_get_grid_cache(mocker):
```

Comment:
It might be good to have a test that verifies that you can get two different grids out. And that the second grid doesn't overwrite the first one.